### ISSUE TITLE
Issues with Mending

### ISSUE THREAD

**Issue by Velosofy** [ViaVersion/ViaBackwards#753]
### '/viaversion dump' Output

Unable to generate, doesn't output anything

### Bug Description

Mending enchantment doesn't really work as expected.

Trying to fix your tools/armor with Mending doesn't really fix the durability, the durability does increase, but it either reset to original or just stop repairing.
Even in the odd chance it "repairs", it doesn't actually repair.

This is the oddness that happens on my server. I was AFK-farming XP, came back and my armor/tools were not repaired.
This is when I first notice the issue.
See [video](https://youtu.be/DofzWWTpdto)
You can see in the video that my armor is not repaired in 1.20.4 client, but when I login with 1.20.6 client, the armor has full durability.

See [video testing](https://youtu.be/f4CnCF5rYk4) for more detail
Left is 1.20.4, Right is 1.20.6

EDIT :
After further testing (more like accidental), in my SMP server (that has same mods with my test server) with 1.20.4 client, the durability value shows the correct one after updating the value ( it's like I have to force-refresh the UI by getting hit to reduce its durability )
![image](https://github.com/ViaVersion/ViaBackwards/assets/61261357/1ec379d2-6b78-45b2-a00b-4f5a047eb68d)

EDIT 2 :
My durability values reset to old values again (not all the armor), seems to be jumping around the place...
I have no idea what's going on now. How can I even diagnose this lmao?

If you like to ask me further question, I'm open for chats in Discord (same handle, in ViaVersion Discord)

### Steps to Reproduce

1. Setup Fabric 1.20.6 server with Via* installed
2. Login with 1.20.4 client
3. Try doing various combination of "repairing tools with Mending"

### Expected Behavior

Mending should work normally.

### Additional Server Info

Client Info:
Vanilla 1.20.4 and 1.20.6

Server Info:
Fabric 1.20.6 (Loader 0.15.11 - Installer 1.0.1) with 4 mods

Server Mods:

fabric-api-0.97.8+1.20.6.jar
ViaFabric-0.4.14+576-dev.jar
ViaBackwards-5.0.0-SNAPSHOT.jar
ViaVersion-5.0.0-SNAPSHOT.jar

### Checklist

- [X] Via plugins are only running on **EITHER** the backend servers (e.g. Paper) **OR** the proxy (e.g. BungeeCord), **not on both**.
- [X] I have included a ViaVersion dump.
- [X] If applicable, I have included a paste (**not a screenshot**) of the error.
- [X] I have tried a build from https://ci.viaversion.com/ and the issue still persists.

**Comment by Barvalg** [comment]
![Error: Invalid or missing dump](https://img.shields.io/badge/Error-Invalid/missing%20dump-red?style=flat-square) </br>

**Comment by FireController1847** [comment]
I'm having the same issue, actually. Connecting a 1.20.1 client to a 1.20.6 server, once I repair my tool completely at least once, then mending will start glitching out. Connecting on a Bedrock client (using Geyser) or using an updated Java Edition client (1.20.6 or higher) will fix this issue, so it seems to be an issue with ViaBackwards. It seems it requires tools that had mending BEFORE updating the server to 1.20.5+ (or when migration to components occurred). This doesn't happen with newly spawned in items. So it's likely a migration issue, something is being handled differently for migrating items that ViaBackwards' code doesn't know about.

To reproduce: Ensure you are using a pre 1.20.5 client (I was able to reproduce on 1.20.1, milage may vary). Launch a pre 1.20.5 (I started on 1.20.1) Spigot server. Enchant some items with mending, then break them a little bit. Update your server to 1.20.5+ (I updated to 1.20.6). Completely repair your items. Suddenly, displaying is broken on the pre 1.20.5 client, indicating an issue with ViaBackwards.

Long story short, +1 on this issue. Glad to know it's only a display bug though, doesn't seem to affect the actual durability of the item on the server's side (it properly repairs on the server, just displays incorrectly on the client).

**Comment by JAdinolfi** [comment]
+1 on this very specific issue, any solution that doesnt involve just remaking tools/armor?

**Comment by kennytv** [comment]
Cannot reproduce with or without an upgraded world, survival or creative, on 5.0.4 snapshot builds
